jablib/src/test/java/org/jabref/logic/citationkeypattern/CitationKeyGeneratorTest.java

```diff
@@ -207,15 +207,19 @@ void makeLabelAndCheckLegalKeys(String bibtexString, String expectedResult) thro
         assertEquals(expectedResult, cleanedKey);
     }
 
-    @Test
-    void firstAuthor() {
-        assertEquals("Newton", generateKey(AUTHOR_FIRSTNAME_INITIAL_LASTNAME_FULL_COUNT_5, "[auth]"));
-        assertEquals("Newton", generateKey(AUTHOR_FIRSTNAME_INITIAL_LASTNAME_FULL_COUNT_1, "[auth]"));
-
-        // https://sourceforge.net/forum/message.php?msg_id=4498555
-        assertEquals("Koening", generateKey(createABibEntryAuthor("K{\\\"o}ning"), "[auth]"));
+    private static Stream<Arguments> firstAuthor() {
+        return Stream.of(
+                Arguments.of(AUTHOR_FIRSTNAME_INITIAL_LASTNAME_FULL_COUNT_5, "Newton"),
+                Arguments.of(AUTHOR_FIRSTNAME_INITIAL_LASTNAME_FULL_COUNT_1, "Newton"),
+                Arguments.of(createABibEntryAuthor("K{\\\"o}ning"), "Koening"),
+                Arguments.of(createABibEntryAuthor(""), "")
+        );
+    }
 
-        assertEquals("", generateKey(createABibEntryAuthor(""), "[auth]"));
+    @ParameterizedTest
+    @MethodSource("firstAuthor")
+    void firstAuthor(BibEntry entry, String expected) {
```

Comment:
expected should always be the first argument, otherwise it's confusing  and inconsistent with other tests
So please adapt this throghout the file